invisible_cities/reco/deconv_functions_test.py

```diff
@@ -0,0 +1,174 @@
+    assert np.allclose(shift_to_bin_centers(np.linspace(-0.05, 1.05, n_interpolation + 1)), sorted(set(out_inter[1][0])))
+    assert np.allclose(shift_to_bin_centers(np.linspace(-0.05, 1.05, n_interpolation + 1)), sorted(set(out_inter[1][1])))
+
+
+def test_deconvolutionInput(data_hdst, data_hdst_deconvolved):
+    ref_interpolation = np.load(data_hdst_deconvolved)
+    hdst              = load_dst(data_hdst, 'RECO', 'Events')
+
+    h = hdst[(hdst.event == 3021916) & (hdst.npeak == 0)]
+    h = h.groupby(['X', 'Y']).Q.sum().reset_index()
+    h = h[h.Q>40]
+
+    interpolator = deconvolutionInput([10., 10.], [1., 1.], 'cubic')
+    inter        = interpolator((h.X, h.Y), h.Q)
+
+    assert np.allclose(ref_interpolation['e_inter'], inter[0])
+    assert np.allclose(ref_interpolation['x_inter'], inter[1][0])
+    assert np.allclose(ref_interpolation['y_inter'], inter[1][1])
+
+
+def test_deconvolve(data_hdst, data_hdst_deconvolved):
+    ref_interpolation = np.load(data_hdst_deconvolved)
+    hdst              = load_dst(data_hdst, 'RECO', 'Events')
+
+    h = hdst[(hdst.event == 3021916) & (hdst.npeak == 0)]
+    z = h.Z.mean()
+    h = h.groupby(['X', 'Y']).Q.sum().reset_index()
+    h = h[h.Q>40]
+
+    deconvolutor = deconvolve(15, 0.01, [10., 10.], [1., 1.], interMethod='cubic')
+
+    x, y   = np.linspace(-49.5, 49.5, 100), np.linspace(-49.5, 49.5, 100)
+    xx, yy = np.meshgrid(x, y)
+    xx, yy = xx.flatten(), yy.flatten()
+
+    psf           = {}
+    psf['factor'] = multivariate_normal([0., 0.], [1.027 * np.sqrt(z/10)] * 2).pdf(list(zip(xx, yy)))
+    psf['xr']     = xx
+    psf['yr']     = yy
+    psf['zr']     = [z] * len(xx)
+    psf           = pd.DataFrame(psf)
+
+    deco          = deconvolutor((h.X, h.Y), h.Q, psf)
+
+    assert np.allclose(ref_interpolation['e_deco'], deco[0].flatten())
+    assert np.allclose(ref_interpolation['x_deco'], deco[1][0])
+    assert np.allclose(ref_interpolation['y_deco'], deco[1][1])
+
+
+def test_richardson_lucy(data_hdst, data_hdst_deconvolved):
+    ref_interpolation = np.load(data_hdst_deconvolved)
+    hdst              = load_dst(data_hdst, 'RECO', 'Events')
+    h = hdst[(hdst.event == 3021916) & (hdst.npeak == 0)]
+    z = h.Z.mean()
+    h = h.groupby(['X', 'Y']).Q.sum().reset_index()
+    h = h[h.Q>40]
+    interpolator = deconvolutionInput([10., 10.], [1., 1.], 'cubic')
+    inter = interpolator((h.X, h.Y), h.Q)
+
+    x, y = np.linspace(-49.5, 49.5, 100), np.linspace(-49.5, 49.5, 100)
+    xx, yy = np.meshgrid(x, y)
+    xx, yy = xx.flatten(), yy.flatten()
+    psf = {}
+    psf['factor'] = multivariate_normal([0., 0.], [1.027 * np.sqrt(z/10)] * 2).pdf(list(zip(xx, yy)))
+    psf['xr']     = xx
+    psf['yr']     = yy
+    psf['zr']     = [z] * len(xx)
+    psf   = pd.DataFrame(psf)
+
+    deco = richardson_lucy(inter[0], psf.factor.values.reshape(psf.xr.nunique(), psf.yr.nunique()).T, 15, 0.0001)
```

Comment:
maybe use `iterations = 50, iter_thr = 1e-4` for a better understanding of the function call